ansible/playbooks/test-kubelet-service-stop/test-kubelet-service-stop.yml

```diff
@@ -0,0 +1,210 @@
+#8.perform Cleanup.
+##########################################################################################
+
+- hosts: localhost
+
+  vars_files:
+    - test-kubelet-service-stop-vars.yml
+
+  tasks:
+
+   - block:
+
+       - include: pre-requisites.yml
+
+
+       - name: Check status of maya-apiserver
+         include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/deploy_check.yml"
+         vars:
+           ns: openebs
+           app: maya-apiserver
+
+
+       - name: Copy the percona files to kube-master
+         copy:
+           src: "{{ item }}"
+           dest: "{{ result_kube_home.stdout }}"
+         delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}"
+         with_items: "{{ percona_files }}"
+
+       - name: Replace volume-claim name with test parameters
+         include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/regex_task.yml"
+         vars:
+           path: "{{ result_kube_home.stdout }}/percona.yaml"
+           regex1: "{{replace_item}}"
+           regex2: "{{replace_with}}"
+         delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}"
+
+
+       - name: Create namespace to deploy application
+         shell: source ~/.profile; kubectl create ns {{ namespace }}
+         args:
+           executable: /bin/bash
+         delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}"
+
+       - name: Create a configmap with the liveness sql script
+         shell: source ~/.profile; kubectl create configmap sqltest --from-file={{ percona_files.1 }} -n {{ namespace }}
+         args:
+           executable: /bin/bash
+         delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+         register: result
+         failed_when: "'configmap' and 'created' not in result.stdout"
+
+       - include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/deploy_task.yml"
+         vars:
+           app_yml: "{{ percona_files.0 }}"
+           ns: "{{ namespace }}"
+
+       - include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/deploy_check.yml"
+         vars:
+            ns: "{{ namespace }}"
+            app: percona
+
+       - name: Wait for 120s to ensure liveness check starts
+         wait_for:
+           timeout: 120
+
+
+       - name: Stop kubelet service in one of the node
+         shell: source ~/.profile; systemctl stop kubelet.service
+         args:
+           executable: /bin/bash
+         register: result
+         become: True
+         delegate_to: "{{groups['kubernetes-kubeminions'].0}}"
+         changed_when: True
+
+
+       - name: check the node is in NotReady state after kubelet service is stopped
+         shell: source ~/.profile; kubectl get nodes | grep NotReady | wc -l
+         args:
+           executable: /bin/bash
+         register: result
+         until: "'1' in result.stdout"
+         delay: 60
+         retries: 5
+         delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+         changed_when: True
```

Comment:
Before stopping kubelet, we can check if all the nodes are ready. This is to ensure that only one node is down after stopping kubelet service